app/controllers/api/items_controller.rb

```diff
@@ -0,0 +1,74 @@
+module API
+  class ItemsController < ApplicationController
+    before_filter :validate_item, only: [:create, :update]
+    before_filter :authorize_token
+    before_filter :my_load_and_authorize_resource, only: [:update, :destroy, :show]
+
+    attr_reader :item
+
+    def create
+      head :created, location: sufia.api_item_path(actor.create_file_from_item)
+    end
+
+    def update
+      actor.update_file_from_item(@file)
+      head :no_content
+    end
+
+    def destroy
+      actor.destroy_file(@file)
+      head :no_content
+    end
+
+    def show
+      head :no_content
+    end
+
+    private
+
+      def my_load_and_authorize_resource
+        @file = GenericFile.find(params[:id])
+
+        unless user.can? :edit, @file
+          return render plain: "#{user} lacks access to #{@file}", status: :unauthorized
+        end
+
+        if @file.arkivo_checksum.nil?
+          return render plain: "Forbidden: #{@file} not deposited via Arkivo", status: :forbidden
+        end
+      rescue ActiveFedora::ObjectNotFoundError
+        return render plain: "id '#{params[:id]}' not found", status: :not_found
+      end
+
+      def actor
+        Sufia::Arkivo::Actor.new(user, item)
+      end
+
+      def token
+        (request.get? || request.delete?) ? params[:token] : item['token']
+      end
```

Comment:
On GET and DELETE, the token is passed as a parameter in the request (because GETs don't have a body, and Arkivo doesn't send DELETEs with a body). With PUT and POST, the token is parsed out of JSON that is in the body of the request.